src/game/Coordinate.java

```diff
@@ -34,23 +34,61 @@ public int getX() {
 	public int getY() {
 		return this.y;
 	}
-	
+
 	public boolean isValid() {
 		return this.valid;
 	}
 
+	public Coordinate down() {
+		return new Coordinate(x, y + 1);
+	}
+	
+	public Coordinate up() {
+		return new Coordinate(x, y - 1);
+	}
+	
+	public Coordinate left() {
+		return new Coordinate(x - 1, y);
+	}
+	
+	public Coordinate right() {
+		return new Coordinate(x + 1, y);
+	}
+	
+	/**
+	 * @return Returns a new Coordinate object that is equivalent to the caller
+	 *         object.
+	 */
```

Comment:
why do you ever need to clone a coordinate? if you do, you probably want to use a copy constuctor instead